Wikipedia/Code/WMFReadingThemesControlsViewController.swift

```diff
@@ -0,0 +1,171 @@
+    }
+    
+    func applyBorder(to button: UIButton) {
+        button.borderColor = UIColor.wmf_blue
+        button.borderWidth = 2
+    }
+    
+    func removeBorderFrom(_ button: UIButton) {
+        button.borderWidth = 0
+    }
+    
+    open func setValuesWithSteps(_ steps: Int, current: Int) {
+        if self.isViewLoaded {
+            self.setValues(0, maximum: steps-1, current: current)
+        }else{
+            maximumValue = steps-1
+            currentValue = current
+        }
+    }
+    
+    func setValues(_ minimum: Int, maximum: Int, current: Int){
+        self.slider.minimumValue = minimum
+        self.slider.maximumValue = maximum
+        self.slider.value = current
+    }
+    
+    override open func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        visible = true
+        
+        if let name = UserDefaults.wmf_userDefaults().string(forKey: "WMFAppThemeName") {
+            switch name {
+            case "standard":
+                applyBorder(to: standardThemeButton)
+            case "light":
+                applyBorder(to: lightThemeButton)
+            case "dark":
+                applyBorder(to: darkThemeButton)
+            default:
+                break
+            }
+        }
+    }
+    
+    @IBAction func brightnessSliderValueChanged(_ sender: UISlider) {
+        UIScreen.main.brightness = CGFloat(sender.value)
+    }
+    
+    @IBAction func fontSliderValueChanged(_ slider: SWStepSlider) {
+        if let delegate = self.delegate, visible {
+            delegate.fontSizeSliderValueChangedInController(self, value: self.slider.value)
+        }
+    }
+    
+    @IBAction func standardThemeButtonPressed(_ sender: Any) {
+        let theme = ["theme": Theme.standard]
+        
+        applyBorder(to: standardThemeButton)
+        removeBorderFrom(lightThemeButton)
+        removeBorderFrom(darkThemeButton)
+        
+        NotificationCenter.default.post(name: Notification.Name(WMFReadingThemesControlsViewController.WMFUserDidSelectThemeNotification), object: nil, userInfo: theme)
+    }
+    
+    @IBAction func lightThemeButtonPressed(_ sender: Any) {
+        let theme = ["theme": Theme.light]
+        
+        applyBorder(to: lightThemeButton)
+        removeBorderFrom(standardThemeButton)
+        removeBorderFrom(darkThemeButton)
+        
+        NotificationCenter.default.post(name: Notification.Name(WMFReadingThemesControlsViewController.WMFUserDidSelectThemeNotification), object: nil, userInfo: theme)
+    }
+    
+    @IBAction func darkThemeButtonPressed(_ sender: Any) {
+        let theme = ["theme": Theme.dark]
+        
+        applyBorder(to: darkThemeButton)
+        removeBorderFrom(standardThemeButton)
+        removeBorderFrom(lightThemeButton)
+        
+        NotificationCenter.default.post(name: Notification.Name(WMFReadingThemesControlsViewController.WMFUserDidSelectThemeNotification), object: nil, userInfo: theme)
```

Comment:
The contents of the `ThemeButtonPressed` methods could be abstracted into a single update method.